src/app/setup-wizard/pre-visit/pre-assessment-setup/pre-assessment-setup.component.ts

```diff
@@ -101,7 +105,29 @@ export class PreAssessmentSetupComponent {
       this.energyEquipmentOptions = equipments;
     });
 
-    this.companyEnergyUnit = this.companyIdbService.getByGUID(this.onSiteVisit.companyId).companyEnergyUnit;
+    this.companySub = this.companyIdbService.selectedCompany.subscribe(_company => {
+      if (_company.guid === this.onSiteVisit.companyId) {
+        this.companyEnergyUnit = _company.companyEnergyUnit;
+      } else {
+        this.companyEnergyUnit = this.companyIdbService.getByGUID(this.onSiteVisit.companyId).companyEnergyUnit;
+      }
+      let filteredAssessments = this.assessments.filter(assessment => this.onSiteVisit.assessmentIds.includes(assessment.guid));
+      filteredAssessments.forEach(async assessment => {
+        await this.calculateEnergyUseCost(assessment);
+      });
+    });
+
+    this.facilitySub = this.facilityIdbService.selectedFacility.subscribe(_facility => {
+      if (_facility.guid === this.onSiteVisit.facilityId) {
+        this.facilityUnitSettings = _facility.unitSettings;
+      } else {
+        this.facilityUnitSettings = this.facilityIdbService.getByGUID(this.onSiteVisit.facilityId).unitSettings;
+      }
+      let filteredAssessments = this.assessments.filter(assessment => this.onSiteVisit.assessmentIds.includes(assessment.guid));
+      filteredAssessments.forEach(async assessment => {
+        await this.calculateEnergyUseCost(assessment);
+      });
```

Comment:
Is there a reason we are calculating the energy use and cost in the subscriptions? We should be calculating them programmatically as the values that effect them change. 

I'm guessing this is to update the use and cost when the facility settings change but we would want to update the use and costs when the facility settings change and add a warning that changing them will effect these values in that page. Relying on the subscription would mean the user has to come to this page for the values to update after the facility settings are changed.

Reply:
Created a new service to update the assessment energy use and cost values and added a warning message for the user.